xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/DatabaseProduct.java

```diff
@@ -156,8 +154,10 @@ public static DatabaseProduct toProduct(String productNameOrJDBCScheme)
         {
             // See documentation above on why we check starts with for DB2
             product = DB2;
-        } else if (isMySQL(productNameOrJDBCScheme)) {
+        } else if (matches(productNameOrJDBCScheme, MYSQL)) {
             product = MYSQL;
+        } else if (matches(productNameOrJDBCScheme, MARIADB)) {
```

Comment:
I'm not a fan of this. It means that any code which does something specific for MySQL/MariaDB will be broken because the type won't be = MYSQL anymore.

To me, MariaDB and MySQL are not different enough from query behavior point of view to justify this very risky change.

Reply:
I totally agree with your feedback.

I tried, in this pull request, to make sure to update all code that might relies on the database being MySQL/MariaDB but I cannot be 100% confident on covering all the code (especially if constant defined in `DatabaseProduct` are not used). So I'm closing this pull request.

In order to solve my issue in Admin Tools, I created [another pull request](https://github.com/xwiki-contrib/application-admintools/pull/36) that modify the code of the extension rather than the code of the platform.
This should be far less risky.